csharp/src/Microsoft.ML.OnnxRuntime/Training/NativeTrainingMethods.shared.cs

```diff
@@ -322,9 +349,9 @@ long seed
             [UnmanagedFunctionPointer(CallingConvention.Winapi)]
             public delegate IntPtr /*(OrtStatus*)*/ DOrtGetProperty(
                                                     IntPtr /*(OrtCheckpointState*)*/ checkpointState,
-                                                    IntPtr /*(const char*)*/ propertyName,
+                                                    byte[] /*(const char*)*/ propertyName,
                                                     IntPtr /*(OrtAllocator*)*/ allocator,
-                                                    out OrtPropertyType propertyType,
+                                                    out long propertyType,
```

Comment:
Why change from passing in OrtPropertyType to having an enum that you cast values to long? 

Can we at least define the OrtPropertyType enum as using the 'long' type to avoid the casts? And if we do that maybe we can still pass in OrtPropertyType here.